rxrabbit/src/main/java/com/meltwater/rxrabbit/impl/SingleChannelConsumer.java

```diff
@@ -150,15 +149,20 @@ private synchronized void startConsuming(Subscriber<? super Message> subscriber,
         );
         InternalConsumer cons;
         String threadNamePrefix = "consume-thread-" + consumerCount;
-        if (consumerRef.get()==null) {
+        if (consumerRef.get() == null) {
             cons = new InternalConsumer(channel, subscriber, closeTimeout, threadNamePrefix, metricsReporter, new AtomicLong(), new AtomicLong());
-        }else{
+        } else {
             cons = new InternalConsumer(consumerRef.get(), threadNamePrefix, channel, subscriber);
         }
-        channel.basicConsume(
-                consumerTag,
-                cons);
         consumerRef.set(cons);
+        try {
+            channel.basicConsume(
+                    consumerTag,
+                    cons);
+        } catch (Exception e) {
+            cons.close();
+            throw e;
+        }
```

Comment:
There's also another thing to consider: we can close InternalConsumer in two places: SingleChannelConsumer.startConsuming or InternalConsumer.createObservable. But it feels that createObservable method should not know about InternalConsumer internals and should not clean up after exceptions, happened in InternalConsumer.